From: Christoph Hellwig <hch@lst.de>
Date: Fri, 11 Mar 2016 17:34:53 +0100
Subject: block: bio_remaining_done() isn't unlikely
Git-commit: 2b885517110cbe8724fef30363778b6284d0a428
Patch-mainline: v4.6-rc1
References: FATE#319965, bsc#964944

We use bio chaining during most I/Os these days due to the delayed
bio splitting.  Additionally XFS will start using it, and there is
a pending direct I/O rewrite also making heavy use for it.  Don't
pretend it's always unlikely, and let the branch predictor do it's
job instead.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.com>
---
 block/bio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/block/bio.c
+++ b/block/bio.c
@@ -1751,7 +1751,7 @@ static inline bool bio_remaining_done(st
 void bio_endio(struct bio *bio)
 {
 again:
-	if (unlikely(!bio_remaining_done(bio)))
+	if (!bio_remaining_done(bio))
 		return;
 
 	/*
